src/main/java/io/github/genomicdatainfrastructure/daam/gateways/RemsApiQueryGateway.java

```diff
@@ -69,4 +77,8 @@ public void checkIfApplicationIsEditableByUser(Long id, String userId) {
             );
         }
     }
+
+    public CatalogueItem retrieveCatalogueItemByResourceId(String resourceId, String userId) {
```

Comment:
hint: for readability arguments of delegated methods should have the same position. This case would be `String userId, String resourceId`.